Filter bad query params

mcumcu opened the pull request:
An error is triggered, when certain characters are in a request query string. This results in an ugly 500 error that is displayed to the client. We can't catch this in the controller, since it occurs in Rack. Instead, we should catch this error in middleware and return a 400 Bad Request response.

- certain characters (e.g. "%") cause errors when parsed by `URI.decode_www_form_component`
- implement a middleware class to parse the request query string and return 400 Bad Request if the request query string fails to parse (instead of 500 Error)

Sentry error: https://sentry.io/organizations/bit-zesty-client-apps/issues/2010483718/?project=78658
Asana: https://app.asana.com/0/1199180449660757/1198831871282498/f

Commit:
Catch bad query params
- these (e.g. "%") cause errors when parsed by URI.decode_www_form_component
- implement middleware class and return 400 Bad Request if the request query string fails to parse (instead of 500 Error)

https://app.asana.com/0/1199180449660757/1198831871282498/f

alexey-klimuk commented on lib/trade_tariff_frontend.rb:
```diff
@@ -102,4 +102,29 @@ def call(env)
+      [
+        400,
+        { "Content-Type" => "application/json" },
+        [{ status: 400, error: "There was a problem with your query: '#{@query_string}'" }.to_json]
```
@mcumcu for api v2 it should be jsonapi compatible

mcumcu replied:
Updated error:
```
{
    "errors": [
        {
            "status": "400",
            "title": "There was a problem with your query",
            "source": {
                "parameter": "query[letter]=%"
            }
        }
    ]
}
```
https://jsonapi.org/format/#errors

Commit:
Return a jsonapi-compatable error message
- "Error objects MUST be returned as an array keyed by errors in the top level of a JSON:API document"
- https://jsonapi.org/format/#errors

Commit:
refactor
